test/plugin_functional/test_suites/core_plugins/dynamic_contract_resolving.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { PluginFunctionalProviderContext } from '../../services';
+
+export default function ({ getService }: PluginFunctionalProviderContext) {
+  const supertest = getService('supertest');
+
+  describe('Dynamic plugin resolving', function describeIndexTests() {
+    it('Plugin A can dynamically resolve plugin B contracts', async () => {
+      await supertest
+        .get('/api/core_dynamic_resolving_a/test')
+        .set('kbn-xsrf', 'anything')
```

Comment:
Added an FTR test to assert that the feature is working, by having 2 plugins dynamically depend on each other to serve their endpoint.